### PR TITLE
WIP: Add a new source for manga scrapping

### PR DESCRIPTION
**Pull request by JohnPCarvalho**
## Rationale:

In order to go further with our application, I'm expanding the functionalities of it by adding a new type of content to scrap.
This is a WIP, and I'll expand this with more code later, but for now:

- Added an external adapter to handle the calls for the source of mangas;
- Added a port to source the HTML delivered by the adapter into our application core;

---
## To do:

- [x] - Find a source for the manga scrapping;
- [x] - Handle the search page of the website;
- [x] - Return the results of the search;

### CODE REVIEW

File: lib/adapter/external/manga_kakalot.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
+defmodule Ewbscp.Adapter.External.MangaKakalot do
+  use Tesla
+
+  plug(Tesla.Middleware.BaseUrl, "https://ww7.mangakakalot.tv")
+  plug(Tesla.Middleware.Headers, [{"authorization", "token xyz"}])
```
Comment (matheusfiorin) [inline lib/adapter/external/manga_kakalot.ex]: Are we going to use this header? 🤔

File: lib/adapter/external/manga_kakalot.ex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+      {:error, reason} ->
+        {:error, reason}
+
+      _ ->
+        {:error, :unknown}
+    end
+  end
+
+  def get_chapter_pages() do
```
Comment (matheusfiorin) [inline lib/adapter/external/manga_kakalot.ex]: The idea is to use this fn in order to fetch the links retrieved by the previous fn?

File: lib/port/manga/manga_kakalot_port.ex [hunk 3]
```diff
@@ -0,0 +1,15 @@
+defmodule Ewbscp.Ports.Manga.MangaKakalot do
+  def return_search_results(manga_search_results) do
+    manga_search_results
+    |> Floki.find(".panel_story_list > .story_item > .story_item_right > h3 > a")
+    |> Enum.map(fn {_, [_, {_, manga_uri}], [manga_title]} ->
+     %{manga_uri: manga_uri, manga_title: manga_title}
```
Comment (matheusfiorin) [inline lib/port/manga/manga_kakalot_port.ex]: We should return a model established in the domain

File: lib/adapter/external/manga_kakalot.ex (hunk 1 above)
Comment (JohnPCarvalho) [reply lib/adapter/external/manga_kakalot.ex]: The BaseUrl one, yes. The others, no. I forgot to take this off.

File: lib/adapter/external/manga_kakalot.ex (hunk 2 above)
Comment (JohnPCarvalho) [reply lib/adapter/external/manga_kakalot.ex]: Initially, yes, but I was thinking about the project structure and thought that maybe this function belongs elsewhere.

File: lib/port/manga/manga_kakalot_port.ex (hunk 3 above)
Comment (JohnPCarvalho) [reply lib/port/manga/manga_kakalot_port.ex]: You're absolutely right. I have some little questions about the architecture. I need to talk with you to solve this questions.

Comment (JohnPCarvalho) [reply lib/port/manga/manga_kakalot_port.ex]: I think I get it. Soon I will push some changes and then you'll see if what I did make sense

File: lib/adapter/external/manga_kakalot.ex [hunk 4]
```diff
@@ -0,0 +1,45 @@
+defmodule Ewbscp.Adapter.External.MangaKakalot do
+  use Tesla
+
+  plug(Tesla.Middleware.BaseUrl, "https://ww7.mangakakalot.tv")
+
+  # returns the page with the search results (this would be a list of links to be presented to the user)
```
Comment (matheusfiorin) [inline lib/adapter/external/manga_kakalot.ex:6]: Let's use [@doc](https://hexdocs.pm/elixir/1.12/writing-documentation.html) on all the external adapters? WDYT?

Comment (matheusfiorin) [reply lib/adapter/external/manga_kakalot.ex:6]: This comment probably is just temporary, but I like the idea of having a `@doc` saying like:
```
Fetches a manga page based on its title from "https://ww7.mangakakalot.tv".

Returns `{:ok, body}`
```

File: lib/adapter/external/manga_kakalot.ex [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+      {:error, reason} ->
+        {:error, reason}
+
+      _ ->
+        {:error, :unknown}
+    end
+  end
+
+  def get_manga_main_page(manga_main_page_uri) do
```
Comment (matheusfiorin) [inline lib/adapter/external/manga_kakalot.ex:20]: Let's think about this: should the **external** adapter **receive** URIs?

Comment (matheusfiorin) [reply lib/adapter/external/manga_kakalot.ex:20]: IMO, we should rethink that. 

Maybe isn't a `port` responsibility to know and send the URI for the adapter 🤔

Comment (matheusfiorin) [reply lib/adapter/external/manga_kakalot.ex:20]: The adapter here is working as a **proxy**, only routing requests and returning its content without any transformation. We should have a model that represents the external content with loyalty, then the `port` will receive it and transform the correspondent **domain model** to which our business logic is applied. Right?

Comment (JohnPCarvalho) [reply lib/adapter/external/manga_kakalot.ex:20]: As we talked earlier today, we're gonna merge this guy and then change all the structure as we want.

**Pull request merged**